File: BerAuto.DataContext/Entities/Rental.cs
namespace BerAuto.DataContext.Entities
{
    public enum RentalStatus
    {
        Pending,
        Approved,
        Rejected,
        PickedUp,
        Returned
    }

    public class Rental
    {
        public int Id { get; set; }
        public int? UserId { get; set; }
        public User User { get; set; }
        public string GuestName { get; set; }
        public string GuestEmail { get; set; }
        public string GuestPhone { get; set; }
        public string GuestAddress { get; set; }
        public int CarId { get; set; }
        public Car Car { get; set; }
        public DateTime RequestDate { get; set; }
        public DateTime? ApprovalDate { get; set; }
        public DateTime? PickupDate { get; set; }
        public DateTime? ReturnDate { get; set; }
        public DateTime From { get; set; }
        public DateTime To { get; set; }
        public RentalStatus Status { get; set; }
        public decimal TotalCost { get; set; }
    }
}

File: lib/models/rental.dart
import 'package:json_annotation/json_annotation.dart';

part 'rental.g.dart';

enum RentalStatus { pending, approved, rejected, pickedUp, returned }

@JsonSerializable()
class Rental {
  final int id;
  final int? userId;
  final String? guestName;
  final String? guestEmail;
  final String? guestPhone;
  final String? guestAddress;
  final int carId;
  final DateTime requestDate;
  final DateTime? approvalDate;
  final DateTime? pickupDate;
  final DateTime? returnDate;
  final RentalStatus status;
  final double totalCost;

  Rental({
    required this.id,
    this.userId,
    this.guestName,
    this.guestEmail,
    this.guestPhone,
    this.guestAddress,
    required this.carId,
    required this.requestDate,
    this.approvalDate,
    this.pickupDate,
    this.returnDate,
    required this.status,
    required this.totalCost,
  });

  factory Rental.fromJson(Map<String, dynamic> json) => _$RentalFromJson(json);
  Map<String, dynamic> toJson() => _$RentalToJson(this);
}